tests/dragonfly/replication_test.py

```diff
@@ -1371,3 +1371,83 @@ async def test_tls_replication(
 
     await c_replica.close()
     await c_master.close()
+
+
+WAIT_FOR_REPLICATION = 2.0
+
+
+@pytest.mark.asyncio
+async def test_replicaof_flag(df_local_factory):
+    # tests --replicaof works under normal conditions
+    master = df_local_factory.create(
+        port=BASE_PORT,
+        proactor_threads=2,
+    )
+
+    # set up master
+    master.start()
+    c_master = aioredis.Redis(port=master.port)
+    await c_master.set("KEY", b"VALUE")
+    db_size = await c_master.dbsize()
+    assert 1 == db_size
+
+    replica = df_local_factory.create(
+        port=BASE_PORT + 1,
+        proactor_threads=2,
+        replicaof=f"localhost:{BASE_PORT}",  # start to replicate master
+    )
+
+    # set up replica. check that it is replicating
+    replica.start()
+    time.sleep(WAIT_FOR_REPLICATION)  # give it time to startup, replication takes longer
+
+    c_replica = aioredis.Redis(port=replica.port)
+
+    dbsize = await c_replica.dbsize()
+    assert 1 == dbsize
+
+    val = await c_replica.get("KEY")
+    assert b"VALUE" == val
+
+
+@pytest.mark.asyncio
+async def test_replicaof_flag_replication_waits(df_local_factory):
+    # tests --replicaof works when we launch replication before the master
+    replica = df_local_factory.create(
+        port=BASE_PORT + 1,
+        proactor_threads=2,
+        replicaof=f"localhost:{BASE_PORT}",  # start to replicate master
+    )
+
+    # set up replica first
+    replica.start()
+    time.sleep(WAIT_FOR_REPLICATION)
```

Comment:
Can we use `wait_available_async` instead of blindly sleeping?
It's better because (1) the tests will take less time to run (and all sleeps accumulate unforutnately), and (2) if for some reason 2s won't be enough (even for 1% of cases), the test will be flaky

Reply:
`wait_for_save` would work here :)